test/avocado/selenium-login.py

```diff
@@ -158,36 +199,201 @@ def test20Login(self):
         elem = self.wait_id('server-name')
 
         elem = self.login("baduser", "badpasswd")
-        elem = self.wait_xpath(
-            "//*[@id='login-error-message' and @style='display: block;']")
-        print elem.text
+        elem = self.wait_id('login-error-message')
         self.assertTrue("Wrong" in elem.text)
 
         elem = self.login()
-        self.wait_iframe("system")
         elem = self.wait_id("content-user-name")
         self.assertEqual(elem.text, user)
+        self.error=False
 
     def test30ChangeTabServices(self):
         self.login()
-        self.wait_iframe("system")
-        self.wait_link('Services').click()
-        self.selectframe("services")
+        self.wait_id("sidebar")
+        self.click(self.wait_link('Services', cond=clickable))
+        self.wait_frame("services")
+        self.wait_id("services-list-enabled")
+        elem = self.wait_text("Socket", cond=clickable)
+        self.click(elem)
+        self.wait_text("udev")
+        self.wait_id("services-list-enabled")
+        elem = self.wait_text("Target", cond=clickable)
+        self.click(elem)
+        self.wait_id("services-list-enabled")
+        self.wait_text("reboot.target")
+        
+        elem = self.wait_text("System Services", cond=clickable)
+        self.click(elem)
+        self.wait_id("services-list-enabled")
+        self.wait_text("dbus.service")
 
-        elem = self.wait_xpath("//*[contains(text(), '%s')]" % "Socket")
-        elem.click()
-        self.wait_xpath("//*[contains(text(), '%s')]" % "udev")
+        self.mainframe()
+        self.error=False
+
+    def test40ContainerTab(self):
+        self.login()
+        self.wait_id("sidebar")
+        self.click(self.wait_link('Containers', cond=clickable))
+        self.wait_frame("docker")
+        if self.wait_xpath("//*[@data-action='docker-start']", fatal=False, overridetry=5, cond=clickable):
```

Comment:
because, if docker is already started, it will take default_try*explicit_wait secs, it is too long, so I made it shorter
.